src/gc/gcinterface.h

```diff
@@ -0,0 +1,499 @@
+    with the GC.
+    ===========================================================================
+    */
+
+    // Finalizes an app domain by finalizing objects within that app domain.
+    virtual BOOL FinalizeAppDomain(AppDomain* pDomain, BOOL fRunFinalizers) = 0;
+
+    // Finalizes all registered objects for shutdown, even if they are still reachable.
+    virtual void SetFinalizeQueueForShutdown(BOOL fHasLock) = 0;
+
+    // Gets the number of finalizable objects.
+    virtual size_t GetNumberOfFinalizable() = 0;
+
+    // Traditionally used by the finalizer thread on shutdown to determine
+    // whether or not to time out. Returns true if the GC lock has not been taken.
+    virtual BOOL ShouldRestartFinalizerWatchDog() = 0;
+
+    // Gets the next finalizable object.
+    virtual Object* GetNextFinalizable() = 0;
+
+    /*
+    ===========================================================================
+    BCL routines. These are routines that are directly exposed by mscorlib
+    as a part of the `System.GC` class. These routines behave in the same
+    manner as the functions on `System.GC`.
+    ===========================================================================
+    */
+
+    // Gets the current GC latency mode.
+    virtual int GetGcLatencyMode() = 0;
+
+    // Sets the current GC latency mode. newLatencyMode has already been
+    // verified by mscorlib to be valid.
+    virtual int SetGcLatencyMode(int newLatencyMode) = 0;
+
+    // Gets the current LOH compaction mode.
+    virtual int GetLOHCompactionMode() = 0;
+
+    // Sets the current LOH compaction mode. newLOHCompactionMode has
+    // already been verified by mscorlib to be valid.
+    virtual void SetLOHCompactionMode(int newLOHCompactionMode) = 0;
+
+    // Registers for a full GC notification, raising a notification if the gen 2 or
+    // LOH object heap thresholds are exceeded.
+    virtual BOOL RegisterForFullGCNotification(uint32_t gen2Percentage, uint32_t lohPercentage) = 0;
+
+    // Cancels a full GC notification that was requested by `RegisterForFullGCNotification`.
+    virtual BOOL CancelFullGCNotification() = 0;
+
+    // Returns the status of a registered notification for determining whether a blocking
+    // Gen 2 collection is about to be initiated, with the given timeout.
+    virtual int WaitForFullGCApproach(int millisecondsTimeout) = 0;
+
+    // Returns the status of a registered notification for determining whether a blocking
+    // Gen 2 collection has completed, with the given timeout.
+    virtual int WaitForFullGCComplete(int millisecondsTimeout) = 0;
+
+    // Returns the generation in which obj is found.
+    virtual unsigned WhichGeneration(Object* obj) = 0;
+
+    // Returns the number of GCs that have transpired in the given generation
+    // since the beginning of the life of the process.
+    virtual int CollectionCount(int generation, int get_bgc_fgc_coutn = 0) = 0;
```

Comment:
>CollectionCount [](start = 16, length = 15)

CollectionCount is also used by AD, and some debug code in finalizer thread proc